pkg/restapi/operation/common/models.go

```diff
@@ -102,3 +104,74 @@ type RegisteredServicesResponse struct {
 	// in: body
 	Names []string `json:"names"`
 }
+
+// SendNewMessageRequest model
+//
+// This is used for operation to send new message
+//
+// swagger:parameters sendNewMessage
+type SendNewMessageRequest struct {
+	// Params for sending new message
+	//
+	// in: body
+	Params *SendNewMessageParams
```

Comment:
Perhaps you could explain this reason for nesting a `SendNewMessageParams` within a `SendNewMessageRequest`. Is this some swagger pattern?

Reply:
Since this is POST message and all params belongs in request body as a single JSON.
Combined them to single struct for 'in:body' annotation

Reply:
if we annote them all 'in:body' separately then, (validation fails)

`- operation "sendNewMessage" has more than 1 body param: ["connection_ID" "message_body" "service_endpoint" "their_did"]` 

we we make only `MessageBody` `in:body` then POST request will look like this,
`curl -X POST "http://localhost:8082/message/send-new-message?connection_ID=asdasd&their_did=asdas" -H "accept: application/json" -H "Content-Type: application/json" -d "{\"msg\": \"msg text is here\"}"`
few params in URL and few in body will make it hard to maintain in future.